packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -3817,6 +3817,165 @@ testWidgetsWithLeakTracking('SnackBarAction backgroundColor works as a Color', (
 
     expect(completer.isCompleted, false);
   });
+
+  testWidgetsWithLeakTracking('SnackbarBehavior.floatingBelowFab is positioned between FAB and BottomBar', (WidgetTester tester) async {
+    const String snackbarAboveContent = 'Snackbar Above';
+    const String snackbarBelowContent = 'Snackbar Below';
+    const String buttonAboveContent = 'Snackbar Above Fab';
+    const String buttonBelowContent = 'Snackbar Below Fab';
+
+    const Duration snackbarDuration = Duration(seconds: 1);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          floatingActionButton: FloatingActionButton(
+            onPressed: () {},
+            child: const Icon(Icons.local_pizza),
+          ),
+          floatingActionButtonLocation: FloatingActionButtonLocation.endTop,
+          body: Builder(
+            builder: (BuildContext context) {
+              return Column(
+                children: <Widget>[
+                  ElevatedButton(
+                    child: const Text(buttonBelowContent),
+                    onPressed: () {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        const SnackBar(
+                          behavior: SnackBarBehavior.floatingBelowFab,
+                          duration: snackbarDuration,
+                          content: Text(snackbarBelowContent),
+                        ),
+                      );
+                    },
+                  ),
+                  ElevatedButton(
+                    child: const Text(buttonAboveContent),
+                    onPressed: () {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        const SnackBar(
+                          behavior: SnackBarBehavior.floating,
+                          duration: snackbarDuration,
+                          content: Text(snackbarAboveContent),
+                        ),
+                      );
+                    },
+                  ),
+                ],
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    Finder fabFinder;
+    Offset fabFinderOffset;
+
+    fabFinder = find.byType(FloatingActionButton);
+    expect(fabFinder, findsOne);
+
+    await tester.tap(find.text(buttonBelowContent));
+    await tester.pumpAndSettle(const Duration(milliseconds: 750));
```

Comment:
@QuncCccccc  I had to add this custom duration (based on some tests above) to fix errors on Linux Web.

> 03:58 +825 ~46: test/material/snack_bar_test.dart: SnackbarBehavior.floatingBelowFab is positioned between FAB and BottomBar                                                                           
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: exactly one matching candidate
  Actual: _TextWidgetFinder:<Found 0 widgets with text "Snackbar Above": []>
   Which: means none were found but one was expected

Does it make sense to you?